### ISSUE TITLE
`None` prompt exits cmdloop

### ISSUE THREAD

**Issue by deargle** [python-cmd2/cmd2#692]
This is a silly edge-case, but if the prompt is None, the cmdloop silently exits. A prompt of `''` does not exit.

I don't want to have a `None` prompt, but I accidentally had one while debugging something else and was scratching my head.
``` python
from cmd2 import Cmd
class MyApp(Cmd, object):

    def __init__(self):
        self.prompt = None
        Cmd.__init__(self)

app = MyApp()
app.cmdloop()
```

Python 2.7, cmd2 v0.8.9

**Comment by kmvanbrunt** [comment]
**cmd2** expects `prompt` to always be a string. The current version of **cmd2** does a bit more processing of `prompt` than v0.8.9, and therefore altering its type will certainly break things. This is to be expected.

If a user actually wants a blank prompt, they can do the following:

```Python
self.prompt = ''
```

**Comment by tleonhardt** [comment]
I did some manual testing and at least with my setup encountered what I consider a legitimate bug that we need to deal with.

I tested on macOS Mojave 10.14.5 using Python 3.7.3 and latest `cmd2` *master* branch from GitHub with Bash shell and iTerm2.  

I tested two cases:
1. prompt set to empty string
    * Caused an infinite loop that got stuck on startup
1. prompt set to None
    * Caused a crash that caused `cmd2` to exit

I think we should add a check somewhere along the lines of the following pseudocode:
```Python3
if not self.prompt or not isinstance(self.prompt, str):
    self.prompt = DEFAULT_PROMPT
```

**Comment by tleonhardt** [comment]
Uh, I was an idiot.  There was never an infinite loop when I set the prompt to an empty string.  I just tested it with an example without an intro banner so it looked like nothing happened.  But then I tried issuing a command like **help** and saw that `cmd2` was running.

**Comment by deargle** [comment]
Yeah similar to popping a non-tty shell through metasploit or whatever.

**Comment by kmvanbrunt** [comment]
The reason setting prompt to `None` fails silently is because `_cmdloop()` is returning within its `finally`. This suppresses the exception caused by the bad prompt. My **script_refactor** branch already fixed the issue in `_cmdloop()` and therefore shows a stack trace. I will close this ticket when that branch merges.

**Comment by kmvanbrunt** [comment]
Closed by #696
